app/scripts/controllers/metametrics-data-deletion/services/services.ts

```diff
@@ -0,0 +1,48 @@
+import getFetchWithTimeout from '../../../../../shared/modules/fetch-with-timeout';
+import { DeleteRegulationAPIResponse } from '../metametrics-data-deletion';
+
+const fetchWithTimeout = getFetchWithTimeout();
+
+export async function createDataDeletionRegulationTask(
+  metaMetricsId: string,
+): Promise<DeleteRegulationAPIResponse> {
+  const segmentSourceId = process.env.SEGMENT_DELETE_API_SOURCE_ID;
+  const segmentRegulationEndpoint = process.env.SEGMENT_REGULATIONS_ENDPOINT;
+
+  try {
+    const response: unknown = await fetchWithTimeout(
+      `${segmentRegulationEndpoint}/regulations/sources/${segmentSourceId}`,
+      {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/vnd.segment.v1+json' },
+        body: JSON.stringify({
+          regulationType: 'DELETE_ONLY',
+          subjectType: 'USER_ID',
+          subjectIds: [metaMetricsId],
+        }),
+      },
+    );
+    return response as DeleteRegulationAPIResponse;
+  } catch (error: unknown) {
+    throw new Error('Analytics Deletion Task Error');
+  }
+}
+
+export async function fetchDeletionRegulationStatus(
+  deleteRegulationId: string,
+): Promise<DeleteRegulationAPIResponse> {
+  const segmentRegulationEndpoint = process.env.SEGMENT_REGULATIONS_ENDPOINT;
+
+  try {
+    const response: unknown = await fetchWithTimeout(
```

Comment:
I wonder if we can type the response in a way other that `unknown` :thinking:

Reply:
The type was explicitly mentioned as it is erroring while converting the `Response` type returned from `fetchWithTimeout ` to DeleteRegulationAPIResponse. We can also typecast while returning the response like this `response as unknown as DeleteRegulationAPIResponse`